src/tests/unit/tests/background/global-action-creators/mock-interpreter.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Interpreter } from 'background/interpreter';
+import { PayloadCallback } from 'common/message';
+import { IMock, It, Mock, Times } from 'typemoq';
+import { DictionaryStringTo } from 'types/common-types';
+
+export class MockInterpreter {
+    private internalMock: IMock<Interpreter>;
+    private messageCallbacksMap: DictionaryStringTo<PayloadCallback<unknown>>;
+
+    constructor() {
+        this.messageCallbacksMap = {};
+        this.internalMock = Mock.ofType<Interpreter>();
+
+        this.internalMock
+            .setup(interpreter => interpreter.registerTypeToPayloadCallback(It.isAny(), It.isAny()))
+            .callback((message, handler) => {
+                this.messageCallbacksMap[message] = handler;
+            });
+    }
+
+    public get object(): Interpreter {
+        return this.internalMock.object;
+    }
+
+    public setupRegisterCallbackForMessage(messageType: string): void {
```

Comment:
Not sure if I follow why this method is necessary with the setup on line 17. What case would you want this for?